Sources/ClientRuntime/Networking/HashFunction.swift

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+import AwsCommonRuntimeKit
+
+enum HashFunction {
+    case crc32, crc32c, sha1, sha256, md5
+
+    static func from(string: String) -> HashFunction? {
+        switch string.lowercased() {
+        case "crc32": return .crc32
+        case "crc32c": return .crc32c
+        case "sha1": return .sha1
+        case "sha256": return .sha256
+        case "md5": return .md5 // md5 is not a valid flexible checksum algorithm
+        default: return nil
+        }
+    }
+
+    var isSupported: Bool {
+        switch self {
+        case .crc32, .crc32c, .sha256, .sha1:
+            return true
+        default:
+            return false
+        }
+    }
+
+    func computeHash(of data: Data) throws -> Data? {
+        switch self {
+        case .crc32:
+            // Should turn this result back into a UInt32 using .toUInt32()
+            return Data(fromUInt32: data.computeCRC32())
+        case .crc32c:
+            // Should turn this result back into a UInt32 using .toUInt32()
+            return Data(fromUInt32: data.computeCRC32C())
+        case .sha1:
+            do {
+                let hashed = try data.computeSHA1()
+                return Data(hashed)
+            } catch {
+                throw ClientRuntime.ClientError.unknownError("Error computing SHA1: \(error)")
```

Comment:
Would it possible to throw a more specific error than `unknownError`?

Reply:
Changed as suggested -- now throws HashError